ziti-fabric/subcmd/set_fingerprint.go

```diff
@@ -0,0 +1,92 @@
+package subcmd
+
+import (
+	"fmt"
+	"github.com/michaelquigley/pfxlog"
+	"github.com/openziti/foundation/storage/boltz"
+	"github.com/spf13/cobra"
+	"go.etcd.io/bbolt"
+	"regexp"
+	"strings"
+)
+
+func init() {
+	Root.AddCommand(setRouterFingerprintCmd)
+}
+
+var setRouterFingerprintCmd = &cobra.Command{
+	Use:   "fingerprint <db-file> <router-id> <fingerprint>",
+	Short: "Manually specify the router client fingerprint, the controller must not be running.",
+	Args:  cobra.ExactArgs(3),
+	Run:   setRouterFingerprint,
+}
+
+func setRouterFingerprint(cmd *cobra.Command, args []string) {
+	dbFile := strings.TrimSpace(args[0])
+	routerId := strings.TrimSpace(args[1])
+	fingerprint := strings.TrimSpace(args[2])
+
+	fingerprint = cleanHexString(fingerprint)
+
+	if dbFile == "" {
+		pfxlog.Logger().Fatalf("<db-file> must specified")
+	}
+
+	if routerId == "" {
+		pfxlog.Logger().Fatalf("<router-id> must specified")
+	}
+
+	if fingerprint == "" {
+		pfxlog.Logger().Fatalf("<fingerprint> must specified and be a valid hex characters")
+	}
+
+	db, err := bbolt.Open(dbFile, 0666, nil)
```

Comment:
i don't suspect you need to close the store but 'should you'?

Reply:
Uhh sure.